tests/tests.c

```diff
@@ -1291,10 +1291,10 @@ static bool generic_syslog_test(const char* sl_name, const char* identity, const
     for (int i = 0; i < runs; i++) {
         /* randomly skip setting process name, identity/category to thoroughly
          * test fallback routines; randomly update the config mid-run. */
-        bool set_procname = getrand_bool((uint32_t)sirtimerelapsed(&timer));
-        bool set_identity = getrand_bool((uint32_t)sirtimerelapsed(&timer));
-        bool set_category = getrand_bool((uint32_t)sirtimerelapsed(&timer));
-        bool do_update    = getrand_bool((uint32_t)sirtimerelapsed(&timer));
+        bool set_procname = getrand_bool(1+(uint32_t)sirtimerelapsed(&timer));
```

Comment:
why is 1+ needed here? is your machine too fast to get random results without it or something?

Reply:
It passes that to the random function, which takes an upper bound.  And the number returned  is usually upper_bound - 1.  So I want to be sure it's >1.

The man page for arc4random_uniform says:

> An upper_bound of 0 doesn't make sense in a call to arc4random_uniform(). Such a call will fail, *and* return 0. Be careful, since that value is not less than upper_bound. In some cases, such as accessing an array, using that value could result in Undefined Behavior.

So I'm just being cautious.

Reply:
Well, sirtimerelapsed is returning milliseconds since the start, so in this case, there is no way it will ever be < 10-100 let alone 1. There may be other spots where we use getrand_bool that are more susceptible to that problem though...

Reply:
And with the adjustment you made in getrand, we're good and don't need a +1 here; this value will always be at least >10. you can log it if you don't believe

Reply:
I'm not gonna budge on this one ;)